include/pybind11/numpy.h

```diff
@@ -542,7 +544,7 @@ class dtype : public object {
             auto name = spec[0].cast<pybind11::str>();
             auto format = spec[1].cast<tuple>()[0].cast<dtype>();
             auto offset = spec[1].cast<tuple>()[1].cast<pybind11::int_>();
-            if (!len(name) && format.kind() == 'V')
+            if ((len(name) == 0u) && format.kind() == 'V')
```

Comment:
Does clang-tidy pick up `unsigned` vs `signed` `0`?

I'm asking because we have this:
https://github.com/pybind/pybind11/blob/a0f862d428acd11dacb344ce9690e420d98cde53/include/pybind11/detail/type_caster_base.h#L255
`status` and `status_instance_registered` or both `uint8_t` (common.h).

But I don't see that picked up in this PR.

Reply:
@rwgk It will try to pick signed or unsigned 0 (although they are bit-equivalent). I can change that to bit equivalent.

Reply:
@rwgk Also, is there a more efficient way to evaluate len(name) is non-zero? Like we had witht he other PR?